container.go

```diff
@@ -35,8 +40,14 @@ type Container interface {
 	SessionID() string                                              // get session id
 	Start(context.Context) error                                    // start the container
 	Terminate(context.Context) error                                // terminate the container
+	Stop(context.Context) error                                     // stop the container
+	Remove(context.Context, bool) error                             // remove the container
 	Logs(context.Context) (io.ReadCloser, error)                    // Get logs of the container
 	Name(context.Context) (string, error)                           // get container name
+	IsRunning(ctx context.Context) (bool, error)                    // is state of container 'running'
+	State(ctx context.Context) (*types.ContainerState, error)       // state of container
```

Comment:
This is a great work. I like it thanks! I am trying to keep the interface as agnostic as possible and the fact that it returns the `*types.ContainerState` tight the interface to Docker and I do not like it for now because the idea is to be able to implement it for other runtimes as well. So for now can you remove the `State` function? The rest sounds good to me!